src/main/java/TaskManager.java

```diff
@@ -0,0 +1,162 @@
+    //response of adding
+    protected String ADD_RES = "Got it. I've added this task:\n";
+    Chatbox chatbox = new Chatbox();
+
+
+    public void saveTask() throws DukeExceptions{
+        String content = "";
+        for(int i = 0;i < Tasks.size();i++){
+            content += Tasks.get(i).getListName();
+            content += "\n";
+        }
+        if(null != content){
+            try{
+                File file = new File("data/Duke.txt");
+                if(!file.exists()){
+                    File dir = new File(file.getParent());
+                    dir.mkdirs();
+                    file.createNewFile();
+                }
+                FileWriter outStream = new FileWriter(file);
+                outStream.write(content);
+                outStream.close();
+            } catch (Exception e){
+                throw new IllegalSavingAction();
+            }
+        }
+
+    }
+    /**
+     * Adds a general task in the list
+     * @param name refers to the name of the task
+     */
+    public void addTask(String name) {
+        Task newTask = new Task(name);
+        Tasks.add(newTask);
+        chatbox.setContent("added: " + name);
+        chatbox.chatboxPrinter();
+    }
+
+    /**
+     * Adds a deadline task to the list
+     * @param name refers to the name of the task
+     * @param by refers to the deadline of the task
+     */
+    public void addDeadline(String name, String by) {
+        Deadline newDeadline = new Deadline(name, by);
+        Tasks.add(newDeadline);
+        int s = Tasks.size();
+        String response = ADD_RES + newDeadline.getListName() +
+                "\n" + "Now you have " + String.valueOf(s) + " tasks in your list.";
+        chatbox.setContent(response);
+        chatbox.chatboxPrinter();
+    }
+
+    /**
+     * Adds an event task to the list
+     * @param name refers to the name of the task
+     * @param at refers to the happening time of the event
+     */
+    public void addEvent(String name, String at) {
+        Event newEvent = new Event(name, at);
+        Tasks.add(newEvent);
+        int s = Tasks.size();
+        String response = ADD_RES + newEvent.getListName() +
+                "\n" + "Now you have " + String.valueOf(s) + " tasks in your list.";
+        chatbox.setContent(response);
+        chatbox.chatboxPrinter();
+    }
+
+    /**
+     * Adds a todo task to the list
+     * @param name refers to the name of the todo task
+     */
+    public void addToDo(String name) {
+        ToDo newToDo = new ToDo(name);
+        Tasks.add(newToDo);
+        int s = Tasks.size();
+        String response = ADD_RES + newToDo.getListName() + "\n" +
+                "Now you have " + String.valueOf(s) + " tasks in your list.";
+        chatbox.setContent(response);
+        chatbox.chatboxPrinter();
+    }
+
+    /**
+     * Lists all tasks in the list with adding time order
+     */
+    public void listTask() {
```

Comment:
You could apply SLAP across this method and [make the happy path more prominent](https://nus-cs2113-ay2122s1.github.io/website/se-book-adapted/chapters/codeQuality.html#make-the-happy-path-prominent).